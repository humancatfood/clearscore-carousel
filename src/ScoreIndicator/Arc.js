import React from 'react';
import { css } from 'emotion';

import { useRAFTween } from './utils';


const arcElementStyle = css({
  position: 'absolute',
  top: 0,
  left: 0,
  width: '100%',
  height: '100%',
});

const gradientID = Math.random().toString().slice(2);


const Arc = ({ value: targetValue }) => {

  if (!targetValue) {
    return null;
  }

  const value = useRAFTween(targetValue);

  const angle = Math.min(Math.max(value, 0), 0.9999);
  const rad = angle * 2 * Math.PI;
  const x1 = 50 + Math.sin(rad) * 50;
  const y1 = 50 - Math.cos(rad) * 50;
  const largeArc = angle > 0.5 ? 1 : 0;

  const d = `M50 0 A50 50 0 ${largeArc} 1 ${x1} ${y1}`;

  return (
    <svg className={arcElementStyle} viewBox="-2 -2 104 104">

      <defs>
        <linearGradient id={gradientID}>
          <stop offset="0" stopColor="#a9cdd6"/>
          <stop offset="100%" stopColor="#B4EDD2"/>
        </linearGradient>
      </defs>

      <path
        d={d}
        fill="none"
        stroke={`url(#${gradientID})`}
      />

    </svg>
  );
};

export default Arc;
