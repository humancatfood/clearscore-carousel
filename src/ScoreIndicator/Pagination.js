import React from 'react';
import { css, cx } from 'emotion';



const paginationStyle = css({
  position: 'absolute',
  bottom: '10%',
  left: '20%',
  width: '60%',
  display: 'flex',
  justifyContent: 'center',
});

const paginationTickStyle = css({
  width: 10,
  height: 10,
  borderRadius: '100%',
  margin: '0 5px',
  cursor: 'pointer',
  border: '2px solid white',
  boxSizing: 'border-box',
  '&.active': {
    backgroundColor: 'white',
  }
});


const Pagination = ({ numPages, currentPageIndex, setCurrentPageIndex }) => {
  if (numPages > 1) {
    return (
      <div className={paginationStyle}>
        {
          Array(numPages).fill(undefined).map((_, index) => (
            <div
              key={index}
              onClick={() => setCurrentPageIndex(index)}
              className={cx(paginationTickStyle, {active: index === currentPageIndex})}
            ></div>
          ))
        }
      </div>
    );
  } else {
    return null;
  }
};

export default Pagination;
