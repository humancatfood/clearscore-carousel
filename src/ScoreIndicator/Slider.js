import React from 'react';
import { css, cx } from 'emotion';





const sliderStyle = css({
  width: '100%',
  height: '100%',
  display: 'flex',
  transition: 'transform 200ms ease-in-out',
});

const slideStyle = css({
  flex: 1,
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  opacity: 0,
  transition: 'opacity 100ms linear',
});

const slideStyleActive = css({
  opacity: 1,
});



const Slider = ({ slides, currentSlideIndex }) => {

  const slideTransformation = css({
    width: `${slides.length * 100}%`,
    transform: `translateX(-${100 * currentSlideIndex / slides.length}%)`,
  });

  return (
    <div className={cx(sliderStyle, slideTransformation)}>
      {
        slides.map((slide, index) => (
          <div key={index} className={cx(slideStyle, {[slideStyleActive]: index === currentSlideIndex})}>
            {
              slide.render()
            }
          </div>
        ))
      }
    </div>
  );
};


export default Slider;
