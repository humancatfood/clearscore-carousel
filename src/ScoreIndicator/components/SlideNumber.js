import React from 'react';

import { css } from 'emotion';



const style = css({
  margin: 5,
  color: '#a9cdd6',
  fontSize: '4rem',
  fontWeight: 'lighter',
});


const SlideNumber = ({ children }) => (
  <h1 className={style}>{ children }</h1>
);

export default SlideNumber;
