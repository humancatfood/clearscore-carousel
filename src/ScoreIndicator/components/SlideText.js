import React from 'react';
import PropTypes from 'prop-types';
import { css, cx } from 'emotion';



export const baseStyle = css({
  margin: 4,
  color: 'white',
});

export const styleHighlight = cx(baseStyle, css({
  color: '#a9cdd6',
}));

const SlideText = ({ highlight=false, className='', ...props }) => (
  <p
    className={cx(
      {
        [styleHighlight]: highlight,
        [baseStyle]: !highlight,
      },
      className,
    )}
    {...props}
  />
);

SlideText.propTypes = {
  highlight: PropTypes.bool,
};

export default SlideText;
