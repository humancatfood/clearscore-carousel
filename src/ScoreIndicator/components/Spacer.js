import React from 'react';



const SlideSpacer = props => (
  <br {...props} />
);

export default SlideSpacer;
