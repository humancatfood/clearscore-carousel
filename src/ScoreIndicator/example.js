import React, { useEffect, useState } from 'react';

import { css } from 'emotion';

import { fetchReportInfo } from './../data';
import ScoreIndicator from './index';



const style = css({
  height: '100%',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
});

const ExampleScoreIndicator = () => {

  const [ data, setData ] = useState(null);
  useEffect(() => {
    fetchReportInfo().then(setData);
  }, []);

  return (
    <div className={style}>
      {
        data ?
          <ScoreIndicator data={data} /> :
          <b>loading..</b>
      }
    </div>
  );

}

export default ExampleScoreIndicator;
