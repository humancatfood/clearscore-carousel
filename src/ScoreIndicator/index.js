import React, { useState } from 'react';
import { css } from 'emotion';

import Pagination from './Pagination';
import Slider from './Slider';



const indicatorStyle = css({
  width: 300,
  height: 300,
  borderRadius: 300,
  maxWidth: '90%',
  backgroundColor: 'rgba(0, 0, 0, 0.6)',
  position: 'relative',
});

const ScoreIndicator = ({ slides }) => {

  const [ currentSlideIndex, setCurrentSlideIndex ] = useState(0);

  console.log('ScoreIndicator:');
  console.log('slides:', slides);
  console.log('slideNum:', currentSlideIndex);

  return (
    <div className={indicatorStyle}>
      <Slider
        slides={slides}
        currentSlideIndex={currentSlideIndex}
      />
      <Pagination
        numPages={slides.length}
        currentPageIndex={currentSlideIndex}
        setCurrentPageIndex={setCurrentSlideIndex}
      />
    </div>
  );
};

export default ScoreIndicator;
