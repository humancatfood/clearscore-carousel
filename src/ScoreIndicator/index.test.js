import React from 'react';
import TestRenderer from 'react-test-renderer';

import ScoreIndicator from './index';
import Spacer from './components/Spacer';
import SlideText from './components/SlideText';
import SlideNumber from './components/SlideNumber';



const textSlide = {
  render: () => (
    <>
      <SlideText>Blabla</SlideText>
    </>
  ),
};

const numberSlide =  {
  render: () => (
    <>
      <SlideNumber>50</SlideNumber>
    </>
  ),
};
const combinedSlide =  {
  render: () => (
    <>
      <SlideNumber>100</SlideNumber>
      <Spacer />
      <Spacer />
      <SlideNumber>50</SlideNumber>
      <Spacer />
      <SlideText>50</SlideText>
      <Spacer />
    </>
  ),
};

describe('ScoreIndicator', () => {

  it('should render', () => {
    const indicator = TestRenderer.create(<ScoreIndicator
      slides={[
        textSlide,
        numberSlide,
        combinedSlide,
      ]}
    />);
    expect(indicator).toMatchSnapshot();
  });

});
