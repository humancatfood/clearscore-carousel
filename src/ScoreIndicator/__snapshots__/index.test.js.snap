// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreIndicator should render 1`] = `
<div
  className="css-qxoi2a"
>
  <div
    className="css-xmumj8"
  >
    <div
      className="css-1c5jdsr"
    >
      <p
        className="css-tqelfc"
      >
        Blabla
      </p>
    </div>
    <div
      className="css-1sghin0"
    >
      <p
        className="css-1r4s34n"
      >
        50
      </p>
    </div>
    <div
      className="css-1sghin0"
    >
      <p
        className="css-1r4s34n"
      >
        100
      </p>
      <br />
      <br />
      <p
        className="css-1r4s34n"
      >
        50
      </p>
      <br />
      <p
        className="css-tqelfc"
      >
        50
      </p>
      <br />
    </div>
  </div>
  <div
    className="css-i65pe3"
  >
    <button
      className="css-1w5ai0h active"
      onClick={[Function]}
      type="button"
    />
    <button
      className="css-1w5ai0h"
      onClick={[Function]}
      type="button"
    />
    <button
      className="css-1w5ai0h"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
